skcosmo/preprocessing/FlexibleScaler.py

```diff
@@ -0,0 +1,242 @@
+import numpy as np
+import sklearn as sk
+from sklearn.base import TransformerMixin, BaseEstimator
+
+
+class StandardFlexibleScaler(TransformerMixin, BaseEstimator):
+    """A more flexible scaler for features and data, that mimics
+    sklearn.preprocessing.StandardScaler make the mean of the
+    columns equal to zero and the variance of each column
+    (if designated) equal to one.
+
+    :param with_mean: If True, center the data before scaling
+    :type with_mean: boolean
+    :param with_std: If True, scale the data to unit variance
+    :type with_std: boolean
+    :param column_wise: If True, normalize each column separately
+    :type column_wise: boolean
+    """
+
+    def __init__(self, *, with_mean=True, with_std=True, column_wise=False):
+        """Initialize StandardFlexibleScaler.
+
+
+        :param with_mean: If True, center the data before scaling
+        :type with_mean: boolean
+        :param with_std: If True, scale the data to unit variance
+        :type with_std: boolean
+        :param column_wise: If True, normalize each column separately
+        :type column_wise: boolean
+        """
+
+        self.with_mean = with_mean
+        self.with_std = with_std
+        self.column_wise = column_wise
+        self.n_samples_seen_ = 0
+
+    def fit(self, X, y=None):
+        """Compute mean and scaling to be applied for subsequent normalization.
+
+        :param X: Matrix
+        :type X: ndarray
+        :param y: ignored
+
+        :return: itself
+        """
+
+        self.n_samples_seen_, self.n_features_ = X.shape
+        self.mean_ = np.zeros(self.n_features_)
+
+        if self.with_mean:
+            self.mean_ = X.mean(axis=0)
+            centred_X = X - self.mean_
+        else:
+            centred_X = X
```

Comment:
So the logic here is that we *always* want a self.mean_ defined, but only want `centred_X` defined if self.with_std (otherwise it is forgotten and never used). Therefore, let's omit `centred_X` here and include the default self.mean_ in the if statement.

```suggestion

        if self.with_mean:
            self.mean_ = X.mean(axis=0)
        else:
            self.mean_ = np.zeros(self.n_features_)
```

Reply:
I slightly change it but save the centered_X because it using for the variance calculating later and can save some computational time. And maybe make the code more readable.